### ISSUE TITLE
Explosive Intent Stacking

### ISSUE THREAD

**Issue by justindbagwell** [bluedudexoo/barbarian-focused-path#2]
The original text is:

`At the 3rd level while raging, your first attack that hits every turn deals an extra d4 of fire damage. If an attack modified in this way hits, your first attack that hits on your next turn deals an additional d4 of fire damage. This effect stacks up to a number of times equal to your strength modifier, and is only lost when your rage ends.`

So if you hit in a turn, the first hit on your next turn does extra damage. This effect cannot stack in practice, even though it says that it can.

Example scenario with 2 attacks per turn:
```
1) Turn 1:
  - Swing and hit, dealing extra d4. 1 charge gained.
  - Swing and hit, dealing normal damage. Zero charges gained. (intent applies to first hit per turn only).
2) Turn 2:
  - Swing and miss, nothing changes.
  - Swing and hit, dealing extra d4 damage and using charge for another d4 damage. 1 charge gained, 1 charge lost.
```

Another example scenario with 2 attacks per turn:
```
1) Turn 1:
  - Swing and hit, dealing extra d4. 1 charge gained.
  - Swing and hit, dealing normal damage. Zero charges gained. (intent applies to first hit per turn only).
2) Turn 2:
  - Swing and miss.
  - Swing and miss.
  - 1 charge lost because the charge from turn 1 only applies to hits in this turn.
2) Turn 2:
  - Swing and miss, nothing changes.
  - Swing and hit, dealing extra d4 damage. 1 charge gained.
```

Additionally "This effect stacks" has an antecedent problem because it could refer to either of the two previous effects.

**Comment by khbsd** [comment]
Yeah, it's kinda tough to figure out proper 5e wording for an effect that doesn't exist, lol.
Hm.

**Comment by khbsd** [comment]
I think something like `for each successive hit modified in this way, your first attack that hits will deal an additional d4 of fire damage.`

**Comment by justindbagwell** [comment]
I think that drastically cleans up the wording, so is better on that front.

The rough math considering a level 8 barb, +5 str, 2 attacks per round, with a greatsword +1, with rage.

Normal damage: 2d6 +5 +1 +2 rage damage. So ~15 damage, making it ~30 damage with the two attacks. At this level, you 
could end up adding 5d4, so another 12.5 damage. If you hit once a round that's a 83% increase. But granted it takes 5 rounds to do so. If it's an average of 2d4, then adding ~5 damage or about +30% with 1 hit, which is still good imo.

So you might consider scaling the max dice with the number of bonus rage damage. So that's 2 dice for level 3-8, 3 for 9-15, and 4 for 16-20. That brings it down a little, caps the high end range, but still lets it scale as the barb levels.

Trying to think of something comparable, and rogue's sneak attack is reasonably close. Sneak attacks are harder to apply, it follows a level progression instead of stat progression, and sneak attacks are once per turn on hit. A level 8 rogue would do: +4d6 or ~14. vs the ~12.5 above.

Also consider Frenzy from Path of the Berserker, which gives a free attack as a bonus action (so ~15 damage at level 8) each round but costs a level of exhaustion. In contrast, explosive intent is "free".

**Comment by khbsd** [comment]
Hmm, what about proficiency bonus? That scales a little bit better, but you don't have that +5d4 until level 13. Of course, at higher levels it's pretty strong, but so is everything then, and at 20 it's only an extra 2.5 damage. Usually I balance with the higher range in mind, but since barbs get a +4 stat increase to con and str at level 20, I figured the end result would be about the same.

Also, Berserker is a little underpowered imo, especially compared to Beast or Zealot or even Ancestral Guardian.

**Comment by justindbagwell** [comment]
Proficiency bonus could work, but consider a level 5 barb. +3d4 (7.5) vs a rogue's +3d6 (10.5).  Even the +2d4 is pretty good.

Normally I'd be whatever about the higher levels because it's kinda whatever (usually issues are at lower levels), but 7d4 is equivalent to the 5d6 sneak attack of a level 10 rogue.

And if berserker is shitty enough that it should be ignored, then compare it with path of the storm instead.

Desert is 2 to 6 damage in a 10 foot AOE but requires your bonus action and can hit teammates. Sea is 1d6 up to 4d6 but requires a bonus action and a saving throw. Whereas this ability starts out at 2d4, would go up to 7d4 or 6d4, and doesn't require a bonus action. Since there isn't a cost for this damage, you're then open to take polearm master, TWF, or shield master (or even GWM) and spam the bonus action.

**Comment by khbsd** [comment]
That's true. I think at level 20 4d4 feels a little underpowered, though. Maybe at level 20 it gets a +1?

**Comment by justindbagwell** [comment]
It's level 20, so ¯\_(ツ)_/¯ I guess. Is fine with 4d4 either way. 

One more thing to consider, since this is additional dice on the melee attack, crits will double the additional dice added (e.g. 4d4 goes to 8d4). 

I didn't realize zealot barb had a very similar ability:
`Starting when you choose this path at 3rd level, you can channel divine fury into your weapon strikes. While you’re raging, the first creature you hit on each of your turns with a weapon attack takes extra damage equal to 1d6 + half your barbarian level. The extra damage is necrotic or radiant; you choose the type of damage when you gain this feature.`

They avoid the massive crit damage by adding half the barb level. Overall the rage damage of d4 dice tracks this somewhat, falling off only at higher levels. Comparing the zealot vs explosive intent: level 3 is 4.5 vs 5. level 6 is 6.5 vs 5. level 9 is 7.5 vs 7.5. level 14 is 10.5 vs 7.5. level 16 is 11.5 vs 10. 

So you get better crits, but scale worse in the level(s) before getting each extra rage damage dice (and at max level) and require multiple rounds to ramp up your damage. As far as balancing goes, it's fairly close. Curious why they gave so a powerful ability to zealot compared to storm barbs though.

**Comment by khbsd** [comment]
Yeah, that subclass is pretty strong. `Defensive Hearth` is essentially Beast's `Call the Hunt` and Zealot's `Zealous Presence`. Initially it was an ability that gave allies resistance to bludgeoning, piercing, and slashing damage while they were in range of the barb, but that would encourage huddling and be too strong lol
